src/pynxtools/nexus/nexus.py

```diff
@@ -581,7 +582,7 @@ def get_single_or_multiple_axes(nxdata, ax_datasets, a_item, ax_list):
                     ax_list.append(nxdata[ax_datasets])
             elif a_item == 0:  # positional determination of the dimension number
                 ax_list.append(nxdata[ax_datasets])
-        else:  # multiple axes are listed
+        elif isinstance(ax_datasets, (list, np.ndarray)):  # multiple axes are listed
```

Comment:
So, now the case, that there is no axis attribute given. At which line of the code, now there is a change in behvior that causes the error?
Which of these variables of "get_single_or_multiple_axes" describes the attribute, which is not present as it is optional?
Maybe I have to reproduce this to understand it - I may also be limited in the knowledge regarding the pynxtools code still....

Reply:
Previous Code:
If no `ax_datasets` is `none` for loop raise an `NoneType` error.
Present Code:
Now the code will check, is it a `list` or `numpy list` otherwise the code will skip that clause and go to the else clause. 

To understand how the empty list `axis` gets handled, you may need to back of the code where the function has been called and how the empty list of `axis` has been handled.

One Clud: The except block says, the parent function can handle a empty list or partially filled.